penguin/web/routes.py

```
@@ -5425,9 +5527,42 @@ async def execute_task_via_runmode(
+        session_id=session_id,
+        conversation_id=conversation_id,
+        agent_id=agent_id,
+    )
+
+
+@router.get("/api/v1/sessions/{session_id}/token-usage")
+async def get_session_token_usage(
+    session_id: str,
+    conversation_id: Optional[str] = Query(default=None),
+    agent_id: Optional[str] = Query(default=None),
+    core: PenguinCore = Depends(get_core),
```

Ruff (0.15.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)